juju_verify/verifiers/result.py

```diff
@@ -171,12 +174,34 @@ def add_partial_result(self, severity: Severity, message: str) -> None:
         self.partials.append(Partial(severity, message))
 
 
-def aggregate_results(*results: Result) -> Result:
-    """Return aggregate value of multiple results."""
-    result_list = list(results)
-    final_result = result_list.pop(0)
-
-    for result in result_list:
-        final_result += result
+def checks_executor(
+        *checks: Union[Callable, Tuple[Callable, Dict[str, Any]]]) -> Result:
+    """Executor that aggregates checks and captures errors.
 
-    return final_result
+    This executor will accept checks as a callable function and will aggregate all
+    their results in order. If the check does not return any input, then the default
+    value is used in the form: `Result(OK, "<check .__ name __> check successful").
+    At the same time, if the check fails on one of the following errors (ActionFailed,
+    CharmException, KeyError, JSONDecodeError), it will be marked as failed with
+    the result in the form:
+    `Result(FAIL, f"{check.__name__} check failed with error: {error}"`.
```

Comment:
Should we also document here a couple of examples (with and without parameters) on how to call this function? This function is core for all verifiers so details on how to use it will immensely help future contributions.